Modules/Scripted/LineProfile/LineProfile.py

```diff
@@ -0,0 +1,638 @@
+from typing import Annotated, Optional
+
+import vtk
+
+import slicer
+from slicer.i18n import tr as _
+from slicer.i18n import translate
+from slicer.ScriptedLoadableModule import *
+from slicer.util import VTKObservationMixin
+from slicer.parameterNodeWrapper import (
+    parameterNodeWrapper,
+    WithinRange,
+)
+
+from slicer import vtkMRMLScalarVolumeNode, vtkMRMLPlotSeriesNode, vtkMRMLTableNode, vtkMRMLMarkupsLineNode
+
+
+#
+# LineProfile
+#
+
+
+class LineProfile(ScriptedLoadableModule):
+    """Uses ScriptedLoadableModule base class, available at:
+    https://github.com/Slicer/Slicer/blob/main/Base/Python/slicer/ScriptedLoadableModule.py
+    """
+
+    def __init__(self, parent):
+        ScriptedLoadableModule.__init__(self, parent)
+        self.parent.title = _("Line Profile")
+        self.parent.categories = [translate("qSlicerAbstractCoreModule", "Quantification")]
+        self.parent.dependencies = []
+        parent.contributors = ["Andras Lasso (PerkLab)"]
+        self.parent.helpText = _("""
+This module computes the intensity profile of a volume along a markups line or curve.
+Notes:
+1. Lines that are on the image boundary (outside the center of border voxels) may be assigned 0 value.
+2. Probed points within the image volume are linearly interpolated among adjacent voxel centers
+and then returned with the same data type as the probed image (e.g. rounded to integers if the image has integer data type).
+    """)
+        self.parent.helpText += self.getDefaultModuleDocumentationLink()
+        self.parent.acknowledgementText = _("""
+This file was originally developed by Andras Lasso (PerkLab) and was partially funded by CCO ACRU.
+""")
+
+
+@parameterNodeWrapper
+class LineProfileParameterNode:
+    """
+    The parameters needed by module.
+
+    inputVolume - The volume to get the intensity profile from.
+    inputLine - The markups line or curve node that defines the path along which the intensity profile is computed.
+    outputTable - The table node that will contain the intensity profile data.
+    lineResolution - The number of points along the line to sample the intensity profile.
+    outputPlotSeries - The plot series node that will contain the intensity profile plot.
+    plotProportionalDistance - If true, the x-axis of the plot will be proportional distance (0-100%).
+    outputPeaksTable - The table node that will contain the peaks of the intensity profile.
+    peakMinimumWidth - The minimum width of the peaks to be detected.
+    heightPercentageForWidthMeasurement - The percentage of the peak height to be used for width measurement.
+    peakIsMaximum - If true, the peaks are considered as maximum values, otherwise as minimum values.
+    """
+    inputVolume: vtkMRMLScalarVolumeNode
+    inputLine: vtkMRMLMarkupsLineNode
```

Comment:
@lassoan parameterNodeWrapper feedback would be helpful here where the input line can be a vtkMRMLMarkupsLineNode or vtkMRMLMarkupsCurveNode, but I'm not sure how to specify that here.